web/src/routes/PoisPage.tsx

```diff
@@ -55,39 +53,57 @@ const moveViewToBBox = (bBox: BBox, defaultVp: MapViewMercatorViewport): MapView
 }
 
 const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: CityRouteProps): ReactElement => {
-  const queryParams = new URLSearchParams(useLocation().search)
-  const { t } = useTranslation('pois')
-  const { featureLocations, pois, poisError, loading } = useFeatureLocations(cityCode, languageCode)
+  const [queryParams, setQueryParams] = useSearchParams()
+  const { data: featureLocations, error: featureLocationsError, loading } = useFeatureLocations(cityCode, languageCode)
+  const mapRef = useRef<MapRef>(null)
+  const selectedFeatureSlug = queryParams.get(locationName)
+  const currentFeature =
+    featureLocations?.find((feature: PoiFeature) => feature.properties.urlSlug === selectedFeatureSlug) ?? null
+  const poi = currentFeature?.properties.poi
+
   const { viewportSmall } = useWindowDimensions()
   const sheetRef = useRef<BottomSheetRef>(null)
   const [feedbackModalRating, setFeedbackModalRating] = useState<FeedbackRatingType | null>(null)
-  const [queryLocation, setQueryLocation] = useState<string | null>(queryParams.get(locationName))
-  const [currentFeature, setCurrentFeature] = useState<PoiFeature | null>(null)
-  const mapRef = useRef<MapRef>(null)
+  const { t } = useTranslation('pois')
 
-  const flyToPoi = (coordinates: Position) => {
-    if (coordinates[0] && coordinates[1] && mapRef.current) {
+  const flyTo = useCallback((coordinates: Position) => {
+    if (!mapRef.current) {
+      // The map ref is not available on initial rendering, therefore retry after a short timeout
+      setTimeout(() => flyTo(coordinates), 1)
```

Comment:
```suggestion
      setTimeout(() => flyTo(coordinates), 0)
```
is sufficient for the retry